### PR TITLE
Re-insert tweet content as HTML

### LINKED ISSUE #39: Syntax highlighting should remove links inside code blocks
See: https://twitter.com/tomasdev/status/953720238784303105

// @tomma5o

### LINKED ISSUE #108: Code block ruins link
See this tweet: https://twitter.com/sindresorhus/status/994307954030526464

The `https://github.com/sindresorhus/sindresorhus` URL is a link when Refined Twitter is turned off. When it's on, it's just text. For some reason it's also showing the URL to the image as text `pic.twitter.com/xZdQSzFLAq`.

// @jorgegonzalez

### PR DESCRIPTION
**Pull request by filipekiss**
Fixes #108 and fixes #39

### CODE REVIEW

**Comment by filipekiss** [comment]
Here it is, working as expected

image

**Comment by jorgegonzalez** [comment]
https://twitter.com/DiscreteQuantum/status/994584722394271744

This is causing links in code blocks to expand even more, but probably because of this bug: https://github.com/sindresorhus/refined-twitter/issues/39

**Comment by filipekiss** [comment]
Fixed :)

image

**Comment by filipekiss** [comment]
So yeah, I just realized there's one more step here:

It's appending the `http` because twitters append's it to anything that remotely looks like a URL but the text is invisible. I'm looking into fixing it right now.

---

Edit: Twitter will append the http and will make the http invisible even if explicitly passed. So every "http://" we have in code will be removed.

**Comment by filipekiss** [comment]
Yeah, just as I tought:

I typed:

```
``javascript
const link="http://google.com"
``
```

Original Tweet:
image

Tweet parsed by Refined Twitter:
image

But this also solves thing like turning `{props.name}` to `{http://props.name}`.

Can we live with the `http` removal?

**Comment by jorgegonzalez** [comment]
I think removing the `http` won't be too controversial.

**Comment by filipekiss** [comment]
Yeah, I'll update the PR and we'll wait for @sindresorhus considerations.

**Review by jorgegonzalez — APPROVED** [review]

File: source/features/code-highlight.js [hunk 1]
```diff
@@ -59,11 +59,27 @@ function highlightCode(md) {
 		</div>
 	);
 }
+
+function cleanupMessyUrls(element) {
+	const elementsToRemove = element.querySelectorAll('.invisible');
+	for (const removableElement of elementsToRemove) {
+		removableElement.remove()
+	}
+	return element
```
Comment (jorgegonzalez) [inline source/features/code-highlight.js:68]: No need to return the element here.

File: source/features/code-highlight.js [hunk 2]
```diff
@@ -59,11 +59,27 @@ function highlightCode(md) {
...
+		tempElement.innerHTML = text;
+		const cleanElement = cleanupMessyUrls(tempElement);
+		frag.append(highlightCode(tempElement.textContent));
 	} else if (text.length > 0) {
-		frag.append(text);
+		const tempElement = document.createElement('div');
+		tempElement.innerHTML = text;
+		for (const child of tempElement.children) {
+			frag.append(child);
+		}
```
Comment (bfred-it) [inline source/features/code-highlight.js]:
```js
frag.append(...tempElement.children);
```

or use domify: https://github.com/sindresorhus/refined-twitter/blob/ad99eb418f9a72fae9077c4d551faa66f3034e8d/source/libs/utils.js#L73

**Comment by sindresorhus** [comment]
Closing this in favor of #113

**Pull request closed without merging**